src/backend/distributed/commands/multi_copy.c

```diff
@@ -465,7 +465,7 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 	 * PG_CATCH. Otherwise, it may be undefined in the PG_CATCH (see sigsetjmp
 	 * documentation).
 	 */
-	shardConnectionHash = CreateShardConnectionHash();
+	copyConnectionHash = CreateShardConnectionHash(MessageContext);
```

Comment:
Just throwing this out there: What if we had a single hash? That seems to be closer to what we want to move towards.